src/lib.rs

```diff
@@ -143,6 +143,33 @@ impl<const N: usize> Ledger<N> {
         }
     }
 
+    /// Check whether the ledger contains a uniform region, and return the
+    /// maximum allowed access for it.
+    pub fn contains(&self, region: Region) -> Option<Access> {
+        assert!(region.start < region.end);
+        assert!(self.region.contains(&region));
```

Comment:
How do I check parameters from userspace, before I run into this panic?

Reply:
Close without answer?

Reply:
I added comment which was meant here (sorry).

Reply:
I agree that some error code makes sense but maybe a single `InvalidRegion` would do.

Reply:
In this case, you can clearly return `None` ... right?

Reply:
Right, I need to fix this. Yes, I can return None here.

Reply:
Hmm, I just meant the second assert... the first was valid.

Reply:
I would first sanitize error codes but idea level agree with. I.e. over time it might make sense to feed the input data as address/length, right?